### ISSUE TITLE
[BUG][Java] useAbstractionForFiles does not work properly on docs .md

### ISSUE THREAD

**Issue by jcolladosp** [OpenAPITools/openapi-generator#19412]
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

`useAbstractionForFiles` changes `File` by `org.springframework.core.io.Resource`  as the return type in the api classes.
This was done in the following MR: https://github.com/OpenAPITools/openapi-generator/pull/13168/files

However in the documentation files generated in markdown, the return type it's still `File`, and the example code doesn't use a `org.springframework.core.io.Resource` 

![image](https://github.com/user-attachments/assets/1cd5b3d6-4319-41df-b4eb-b0566337a985)

##### openapi-generator version

7.8.0

##### Steps to reproduce

Use property `useAbstractionForFiles = true`

##### Related issues/PRs

https://github.com/OpenAPITools/openapi-generator/issues/10665

##### Suggest a fix

I think the documentation files should use `org.springframework.core.io.Resource` if `useAbstractionForFiles=true`
And the example code snippet should be modified accordingly

**Comment by wing328** [comment]
yes can you please file a PR accordingly to update the mustache template for the auto-generated documentation?
